### PR TITLE
Remove ListServiceHooks

### LINKED ISSUE #2916: ListServiceHooks appears to no longer be supported
Client.ListServiceHooks appears to be no longer supported by github. The backing endpoint is GET https://api.github.com/hooks. That is currently 404 and not mentioned in any of the published openapi descriptions for enterprise server.

I think it can be safely removed from go-github along with the supporting ServiceHook struct.

I can create a PR for this.

### PR DESCRIPTION
**Pull request by WillAbides**
closes #2916

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @WillAbides !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
